frontend/packages/knative-plugin/src/components/add/eventSource-validation-utils.ts

```diff
@@ -65,9 +65,24 @@ export const sourceDataSpecSchema = yup
   });
 
 export const eventSourceValidationSchema = yup.object().shape({
-  project: projectNameValidationSchema,
-  application: applicationNameValidationSchema,
-  name: nameValidationSchema,
-  sink: sinkServiceSchema,
-  data: sourceDataSpecSchema,
+  project: yup.object().when('editorType', {
+    is: EditorType.Form,
+    then: projectNameValidationSchema,
+  }),
+  application: yup.object().when('editorType', {
+    is: EditorType.Form,
+    then: applicationNameValidationSchema,
+  }),
+  name: yup.string().when('editorType', {
```

Comment:
can have validation based on `type` itself  and remove `editorType` field  WDYT?

```
const ownEvenstSources = Object.keys(EventSources);
export const eventSourceValidationSchema = yup.object().shape({
  project: yup.object().when('type', {
    is: (type) => ownEvenstSources.includes(type),
    then: projectNameValidationSchema,
  }),
.....
})
```

Reply:
Or simply add `.when` to the object and add all the properties based on that one check.
```suggestion
export const eventSourceValidationSchema = yup.object().when('type', {
  is: (type) => Object.keys(EventSources).includes(type),
  then: yup.object().shape({
    project: projectNameValidationSchema,
    application: applicationNameValidationSchema,
    name: nameValidationSchema,
    sink: sinkServiceSchema,
    data: sourceDataSpecSchema,
  }),
  otherwise: yup.object().shape({
    data: sourceDataSpecSchema,
  }),
});
```

Reply:
`is: (type) => Object.keys(EventSources).includes(type),` was throwing error with this message - `type is not a function`
approached it using yup.lazy(). PTAL